### PR TITLE
feat(consensus): beacon consensus builder

### LINKED ISSUE #1093: Create `BeaconConsensus` builder
### Describe the feature

1. Create a simple beacon consensus builder
 https://github.com/paradigmxyz/reth/blob/f771e23f9a3e4238bb413a55f02d1536126a0ea1/bin/reth/src/node/mod.rs#L156

2. Decouple and extract the sender part of the forkchoice state channel from the `BeaconConsensus`.

### PR DESCRIPTION
**Pull request by rkrasiuk**
Closes #1093 

Create a basic `BeaconConsensusBuilder` and decouple forkchoice state notifier from `BeaconConsensus`

### CODE REVIEW

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use clap::{crate_version, Parser};
 use eyre::Context;
 use fdlimit::raise_fd_limit;
 use futures::{stream::select as stream_select, Stream, StreamExt};
-use reth_consensus::beacon::BeaconConsensus;
+use reth_consensus::beacon::BeaconConsensusBuilder;
```
Comment (mattsse) [inline bin/reth/src/node/mod.rs:14]: I'd like to use all builders like `BeaconConsensus::builder()` instead of importing the builder

**Review by mattsse — APPROVED** [review]
smol nit that we could apply to all builder types.

**Review by onbjerg — CHANGES_REQUESTED** [review]
Sorry, I think you misunderstood the comment linked in #1093 - I want a consensus engine builder that returns an `Arc<dyn Consensus>`, not a builder for `BeaconConsensus`.

The chain spec is going to include parameters for different consensus engines (e.g. clique) and we should be able to, at runtime, construct the correct consensus engine from this info.

**Comment by rkrasiuk** [comment]
@onbjerg we don't have a separate issue for consensus factory, do we? i'll create it. anyway, `BeaconConsensusBuilder` is useful for decoupling notifier and consensus impl

**Comment by onbjerg** [comment]
@rkrasiuk We don't

**Comment by rkrasiuk** [comment]
@onbjerg merging this PR as is then and will create an issue

**Pull request merged**